### DISCUSSION TITLE
小白，pool里有个地方不懂，感觉要出现死锁样子

### DISCUSSION THREAD

**Discussion by ogios** [panjf2000/ants#301] [Q&A]
如果 `retrieveWorker` 方法在阻塞且worker满了的状态下进了 `retry` 中，没有获取到worker就会一直在 `retry` 中循环而无法调用`lock.Unlock` 释放锁，是这样吗？
但是 `revertWorker` 中需要锁才能进入 `insert` 然后添加worker，于是就一直卡在 [`lock.Lock()`](https://github.com/panjf2000/ants/blob/v2.8.2/pool.go#L379) 上。

除非有一个别的地方在不调用 `p.lock.Lock()` 的情况下直接调用 `p.lock.Unlock()` 强行释放。

我不知道我说的对不对，恳请各位大佬指点以下，我这里一直不太能看懂。

**Comment by panjf2000** [comment]
有很多地方都会释放锁，你说的一直循环不释放锁我不明白，你贴下代码解释下？
